File: py/leetcode-maxchoc.py
class Solution:
    def buyChoco(self, prices: List[int], money: int) -> int:
        l = 1000
        sl = 999

        for p in prices:
            if p < l:
                sl = l
                l = p
            elif p < sl:
                sl = p

        return money - l - sl if money - l - sl >=0 else money
        

File: rs/leetcode-maxchoc.rs
impl Solution {
    pub fn buy_choco(prices: Vec<i32>, money: i32) -> i32 {
        if prices.len() < 2 {
            return money;
        }

        let mut h = std::i32::MAX;
        let mut sh = std::i32::MAX;

        for i in 0..prices.len() {
            if prices[i] < h {
                sh = h;
                h = prices[i];
            } else if prices[i] < sh {
                sh = prices[i];
            }
        }

        if money - h - sh < 0 {
            return money;
        }
        money - h - sh
    }
}
